src/main/resources/META-INF/resources/primefaces/rating/rating.js

```diff
@@ -92,18 +92,32 @@ PrimeFaces.widget.Rating = PrimeFaces.widget.BaseWidget.extend({
      * @param {number} value New rating value to set (number of starts selected).
      */
     setValue: function(value) {
+        if(this.jq.hasClass('ui-state-disabled') || this.jqInput.is('[readonly]')) {
```

Comment:
maybe add a isDisabled and isReadonly method for it?

Reply:
Great idea. Added!